src/test/java/io/appium/java_client/android/LogEventTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.android;
+
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import io.appium.java_client.serverevents.CommandEvent;
+import io.appium.java_client.serverevents.ServerEvent;
+import io.appium.java_client.serverevents.ServerEvents;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+public class LogEventTest extends BaseAndroidTest {
+
+    @Test
+    public void verifyLoggingCustomEvents() {
+        driver.logEvent("appium", "funEvent");
+        ServerEvents events = driver.getEvents();
+        boolean hasCustomEvent = false;
+        boolean hasCommandName = false;
+        for (ServerEvent event : events.events) {
```

Comment:
consider using streams + filter + anyMatch

Reply:
👍

Reply:
java is actually much more pleasant with these functional bits.